### PR TITLE
Adds shock wave shader

### PR DESCRIPTION
**Pull request by vero5123**
## About the PR
WIP needs to be tweaked, adds a shockwave overlay.

## Why / Balance
visually pleasing

## Technical details
Added a new shader along with an overlay for shock waves. As well as a new shockwave component on the explosive effect entity. 

## Media
https://github.com/RMC-14/RMC-14/assets/73014819/79182096-2860-4b09-83c6-056ba6f45ce8

- [x] I have added screenshots/videos to this PR showcasing its changes ingame

:cl:
- add: adds a shockwave overlay effect.

### CODE REVIEW

File: Resources/Prototypes/Shaders/shaders.yml
Comment (QuietlyWhisper) [inline Resources/Prototypes/Shaders/shaders.yml:1]: this should be under `Prototypes/_RMC14/Shaders`

Also ensure all new content IDs are `RMC...`

File: Resources/Textures/Shaders/shock_wave.swsl
Comment (QuietlyWhisper) [inline Resources/Textures/Shaders/shock_wave.swsl:1]: this should be under `Textures/_RMC14/Shaders`

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Throwable/grenades.yml
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Throwable/grenades.yml:1]: Ensure all new content IDs are `RMC...` and not `CM...`

File: Content.Shared/_RMC14/Explosion/CMExplosionShockWaveComponent.cs
Comment (QuietlyWhisper) [inline Content.Shared/_RMC14/Explosion/CMExplosionShockWaveComponent.cs:1]: Ensure all new content IDs/Filenames are `RMC...` and not `CM...`

File: Content.Client/_RMC14/Explosion/CMExplosionShockWaveOverlay.cs
Comment (QuietlyWhisper) [inline Content.Client/_RMC14/Explosion/CMExplosionShockWaveOverlay.cs:1]: Ensure all new content IDs/Filenames are `RMC...` and not `CM...`

File: Content.Client/Entry/EntryPoint.cs
Comment (QuietlyWhisper) [inline Content.Client/Entry/EntryPoint.cs:1]: Ensure all new content IDs/Filenames are `RMC...` and not `CM...`

File: Content.Client/_RMC14/Explosion/CMExplosionShockWaveOverlay.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
+using Content.Shared._RMC14.Explosion.Components;
+using Robust.Client.Graphics;
+using Robust.Shared.Enums;
+using Robust.Shared.Prototypes;
+using System.Collections.Generic;
+using System.Numerics;
+
+namespace Content.Client._RMC14.Explosion
```
Comment (QuietlyWhisper) [inline Content.Client/_RMC14/Explosion/CMExplosionShockWaveOverlay.cs:8]: `namespace Content.Client._RMC14.Explosion;`

you can put a semicolon here and drop a whole line of indentation across the entire file

File: Content.Client/_RMC14/Explosion/CMExplosionShockWaveOverlay.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        public CMExplosionShockWaveOverlay()
+        {
+            IoCManager.InjectDependencies(this);
+            _shader = _prototypeManager.Index<ShaderPrototype>("ShockWave").Instance().Duplicate();
+        }
+
+         private readonly Vector2[] _positions = new Vector2[MaxCount];
+        private readonly float[] _falloffPower = new float[MaxCount];
+        private readonly float[] _sharpness = new float[MaxCount];
```
Comment (QuietlyWhisper) [inline Content.Client/_RMC14/Explosion/CMExplosionShockWaveOverlay.cs:35]:
```suggestion
        private readonly Vector2[] _positions = new Vector2[MaxCount];
        private readonly float[] _falloffPower = new float[MaxCount];
        private readonly float[] _sharpness = new float[MaxCount];
```
empty space

**Review by QuietlyWhisper — CHANGES_REQUESTED** [review]
Mostly just naming conventions

**Comment by vero5123** [comment]
This needs to be done in a way such that it isn't an eye sore, so I'm going to have to find a better solution as it isn't always as pretty as shown in the demo upload.

**Comment by vero5123** [comment]
newest changes should be better, I guess i'll just wait for maintainer input.

**Comment by vero5123** [comment]
so this was hefa before:

https://github.com/RMC-14/RMC-14/assets/73014819/697857d4-c8fc-4a06-abae-f8260a2734d4

and this is hefa now:

https://github.com/RMC-14/RMC-14/assets/73014819/6e9ed644-6c78-45b3-8450-7e9dca892940

**Comment by vero5123** [comment]
more circular explosions

https://github.com/RMC-14/RMC-14/assets/73014819/415ae3f4-b83d-43a2-b36c-780e8e1bbdda

File: Content.Server/Explosion/EntitySystems/ClusterGrenadeSystem.cs [hunk 3]
```diff
@@ -122,8 +123,9 @@ public override void Update(float frameTime)
 
                 var clusterEv = new CMClusterSpawnedEvent(_spawned);
                 RaiseLocalEvent(uid, ref clusterEv);
-                // delete the empty shell of the clusterbomb
-                Del(uid);
+                //del or qdel causes lag out the ass for shrapnel type clusters.
+                var despawn = AddComp<TimedDespawnComponent>(uid);
+                despawn.Lifetime = 0.01f;
```
Comment (QuietlyWhisper) [inline Content.Server/Explosion/EntitySystems/ClusterGrenadeSystem.cs:128]: Can you explain this change? We're not sure how this helps performance

(Also if it does help maybe it's a good upstream change as well?)

**Review by QuietlyWhisper — CHANGES_REQUESTED** [review]

File: Content.Server/Explosion/EntitySystems/ClusterGrenadeSystem.cs (hunk 3 above)
Comment (vero5123) [reply Content.Server/Explosion/EntitySystems/ClusterGrenadeSystem.cs:128]: discussed on discord, it was an issue with the component storing the deleted uid in one of the automatically networked fields

**Review by DrSmugleaf — APPROVED** [review]
Thanks for the PR the effect looks great!

**Pull request merged**